### ISSUE TITLE
How do you reproduce the FID for DIV2K SISR task?

### ISSUE THREAD

**Issue by yuanzhi-zhu** [Hammour-steak/GOUB#11]
Thanks for your great work!

How do you evaluate FID on the DIV2K validation set (100 images)?
do you center crop the validation images into 256*256 patches and evaluate the downsample and bicubic upsampled images with your model?
and for which two folders do you compute the FID?
There are several details missing to reproduce the results, would be great if you can help explain the FID evaluation process, or release a python script to reproduce the result.

**Comment by Hammour-steak** [comment]
Thank you for your interest in our work!

During validation, center cropping is **not** applied. Note that the LQ input resolution for the model should **match** the GT resolution. This is achieved by preprocessing the LQ image with bicubic interpolation. For instance, if the GT is 4x4, the corresponding LQ is 1x1, so you should first upsample the LQ to 4x4 using bicubic interpolation before training and inference. When calculating FID, it should be computed between the **generated images** and the **GT** folder.

Additionally, center cropping is **only** used during training. And for all tasks, the original image is used for validation.

We hope this information is helpful to you.

**Comment by yuanzhi-zhu** [comment]
@Hammour-steak 
thanks for your reply, I have two follow up questions
which two sets do you use to calculate the FID? do you evaluate on the valid set with 100 images and compute the fid with 800 training set images?
do you use tile inference to process the image or take the whole image as input to the network?

**Comment by Ly403** [comment]
> Thank you for your interest in our work!
> 
> During validation, center cropping is **not** applied. Note that the LQ input resolution for the model should **match** the GT resolution. This is achieved by preprocessing the LQ image with bicubic interpolation. For instance, if the GT is 4x4, the corresponding LQ is 1x1, so you should first upsample the LQ to 4x4 using bicubic interpolation before training and inference. When calculating FID, it should be computed between the **generated images** and the **GT** folder.
> 
> Additionally, center cropping is **only** used during training. And for all tasks, the original image is used for validation.
> 
> We hope this information is helpful to you.

Hi. I'm curious about the center crop process for the DIV2K dataset. Did you only crop the images using the instructions in the `LQGT dataset`([line 109](https://github.com/Hammour-steak/GOUB/blob/4724d1f573bc812c9e6d13f3a929014ce43ee909/codes/data/LQGT_dataset.py#L109)), or did you also preprocess the images by extracting sub-images like the operations in [basicsr](https://github.com/XPixelGroup/BasicSR/blob/master/docs/DatasetPreparation.md#div2k). I mean **in the training process**.

Thanks a lot!

**Comment by Hammour-steak** [comment]
> > Thank you for your interest in our work!
> > During validation, center cropping is **not** applied. Note that the LQ input resolution for the model should **match** the GT resolution. This is achieved by preprocessing the LQ image with bicubic interpolation. For instance, if the GT is 4x4, the corresponding LQ is 1x1, so you should first upsample the LQ to 4x4 using bicubic interpolation before training and inference. When calculating FID, it should be computed between the **generated images** and the **GT** folder.
> > Additionally, center cropping is **only** used during training. And for all tasks, the original image is used for validation.
> > We hope this information is helpful to you.
> 
> Hi. I'm curious about the center crop process for the DIV2K dataset. Did you only crop the images using the instructions in the `LQGT dataset`([line 109](https://github.com/Hammour-steak/GOUB/blob/4724d1f573bc812c9e6d13f3a929014ce43ee909/codes/data/LQGT_dataset.py#L109)), or did you also preprocess the images by extracting sub-images like the operations in [basicsr](https://github.com/XPixelGroup/BasicSR/blob/master/docs/DatasetPreparation.md#div2k). I mean **in the training process**.
> 
> Thanks a lot!

Hey Ly403, we crop the images using the instructions in the LQGT dataset, the training code is all default.
